crates/noirc_frontend/src/node_interner.rs

```diff
@@ -56,7 +56,7 @@ pub struct NodeInterner {
     //
     // Map type aliases to the actual type.
     // When resolving types, check against this map to see if a type alias is defined.
-    type_aliases: HashMap<TyAliasId, Shared<TypeAliasTy>>,
+    type_aliases: HashMap<TypeAliasId, TypeAliasType>,
```

Comment:
This can be just a `Vec<TypeAliasType>` indexed by the TypeAliasId which is just an index into the Vec - like the `definitions` field.